### PR TITLE
BugFix: Prevent null exception if block for minting is null(#803)

### PR DESCRIPTION
**Pull request by kogot**
Checking if GetBlock() return null then set hashBlock as a uint256.Zero

### CODE REVIEW

**Comment by Aprogiena** [comment]
We should not use zero there, we should just skip this UTXO.

File: src/Stratis.Bitcoin.Features.Miner/PosMinting.cs [hunk 1]
```diff
@@ -500,11 +500,12 @@ public void StopStake()
                         utxoStakeDescription.TxOut = utxo;
                         utxoStakeDescription.OutPoint = new OutPoint(set.TransactionId, infoTransaction.Transaction.Index);
                         utxoStakeDescription.Address = infoTransaction.Address;
-                        utxoStakeDescription.HashBlock = this.chain.GetBlock((int)set.Height).HashBlock;
+                        utxoStakeDescription.HashBlock = this.chain.GetBlock((int)set.Height)?.HashBlock;
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Miner/PosMinting.cs]: This is a good I would suggest to put the line`var blockHash = 
 this.chain.GetBlock((int)set.Height)?.HashBlock;` before the `if`
then use it as part of the condition 
` if ((utxo != null) && (utxo.Value > Money.Zero) && (blockHash != null))`

Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.Miner/PosMinting.cs]: yeah that would be even better

Comment (kogot) [reply src/Stratis.Bitcoin.Features.Miner/PosMinting.cs]: Agree, it will be better.

**Comment by kogot** [comment]
@dangershony  can you please review it once you have chance

**Pull request merged**